### PR TITLE
Switch to using Oasis instances instead of global.

### PR DESCRIPTION
**Pull request by hjdivad**
This is important to getting the inline adapter to work without a specialized loader.

### CODE REVIEW

File: lib/oasis/connect.js [hunk 1]
```diff
@@ -86,19 +82,48 @@ export function registerHandler(capability, options) {
...
+  a_forEach.call(capabilities, function(capability, i) {
+    var handler = oasis.handlers[capability],
+        port = new PostMessagePort(eventPorts[i]);
+
+    if (handler) {
+      Logger.log("Invoking handler for '" + capability + "'");
+
+      RSVP.resolve(handler.setupCapability(port)).then(function () {
+        port.start();
+      });
```
Comment (stefanpenner) [inline lib/oasis/connect.js:104]: missing `fail(RSVP.rethrow)`

File: vendor/loader.js [hunk 2]
```diff
@@ -8,8 +8,10 @@ if (typeof define !== 'function' && typeof requireModule !== 'function') {
       registry[name] = { deps: deps, callback: callback };
     };
 
-    requireModule = function(name) {
-      if (seen[name]) { return seen[name]; }
+    requireModule = function(name, sessionId) {
```
Comment (stefanpenner) [inline vendor/loader.js]: im not sure you are using the session aspect of the loader.

File: dist/oasis.amd.js [hunk 3]
```diff
@@ -607,6 +615,138 @@ define("oasis/iframe_adapter",
...
+      initializeSandbox: function(sandbox) {
+        sandbox.el = document.createElement('div');
+        var channel = new PostMessageMessageChannel();
+
+        this.port = channel.port1;
+
+        sandbox.sandbox = new InlineSandbox({
+          port: channel.port2
+        });
+        sandbox.sessionId = sessionId;
```
Comment (stefanpenner) [inline dist/oasis.amd.js]: is this still being used, or just remnants from our original hacks?

Comment (hjdivad) [inline dist/oasis.amd.js]: i need to rebuild; actually i'm still missing one more place where we're using globals that i need to fix

Comment (hjdivad) [inline dist/oasis.amd.js]: this PR shouldn't include the inline adapter

Comment (hjdivad) [inline dist/oasis.amd.js]: it should only be fixing the instance issue so the inline adapter does not require hacks

**Pull request merged**